### PR TITLE
Bump jenkins baseline to 2.319.3, parent pom and needed dependencies

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
There were failures in PCT runs against the upcoming 2.375.1 LTS due to out of date test dependencies.

Reproduced by `mvn test -Djenkins.version=2.375`  

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,9 +37,9 @@
     <revision>2.28</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.235.5</jenkins.version>
+    <jenkins.version>2.319.3</jenkins.version>
     <java.level>8</java.level>
-    <configuration-as-code.version>1.35</configuration-as-code.version>
+    <configuration-as-code.version>1569.vb_72405b_80249</configuration-as-code.version>
```
Comment (jglick) [inline pom.xml:42]: Should be taken from BOM.

File: pom.xml [hunk 2]
```diff
@@ -78,7 +78,7 @@
     <dependency>
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>caffeine-api</artifactId>
-      <version>2.9.0-17.v79e9099a51e3</version>
+      <version>2.9.3-65.v6a_47d0f4d1fe</version>
```
Comment (jglick) [inline pom.xml:81]: IIRC this as well.

File: pom.xml [hunk 3]
```diff
@@ -37,9 +37,9 @@
     <revision>2.28</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.235.5</jenkins.version>
+    <jenkins.version>2.319.3</jenkins.version>
     <java.level>8</java.level>
```
Comment (jglick) [inline pom.xml:41]:
```suggestion
```

**Review by jglick — COMMENTED** [review]
Use `bom`, and delete any `<version>`s that are no longer required.

**Review by basil — CHANGES_REQUESTED** [review]
Based on https://ci.jenkins.io/job/Plugins/job/active-directory-plugin/job/PR-134/3/console I think this will also need an update of the Docker image used in the functional tests as described in https://github.com/jenkinsci/active-directory-plugin/pull/134#issuecomment-1259777266.

**Comment by raul-arabaolaza** [comment]
I am now using the BOM, I will check testcontainers for that it failure

**Review by jglick — APPROVED** [review]

**Review by rsandell — APPROVED** [review]

**Comment by raul-arabaolaza** [comment]
Anyone knows the source of the `fbelzunc/ad-build-container-with-docker-fixtures` image? I have been unable to find the dockerfile in dockerhub and https://github.com/jenkinsci/active-directory-plugin/blob/master/src/test/resources/hudson/plugins/active_directory/docker/TheFlintstonesTest/TheFlintstones/Dockerfile is not it, the cmd instructions are different

**Comment by jglick** [comment]
Did you ask @fbelzunc?

For the future, I would recommend using Testcontainers in the mode that you can build an image from source in the project. Slower (depending on cache setup) but easier to maintain and understand.

**Comment by raul-arabaolaza** [comment]
Yes, but he is not available until Monday, that is why I asked here just in case. Totally agree with using testcontainers building from source, that is why I want the source Dockerfile

**Comment by fbelzunc** [comment]
I think it is https://github.com/fbelzunc/active-directory-plugin-build-container-with-docker-fixtures

**Comment by raul-arabaolaza** [comment]
I have been reviewing the Dockerfile that Felix linked and I do not believe I can replace it with test containers or any other thing.

IIUC that is used to run integration tests in a system with a known host name, the tests themselves use dockerfixtures that could be replaced with testcontainers, but I see no way to ensure running those tests in a controlled hostname using testcontainers, more than probably due to my lack of knowledge. Having the tests consult the hostname of the system where they are run looks less than ideal to me and may introduce portability issues.

Doing a big refactoring of the tests to address this issues is totally out of the scope of this pr and my knowledge, so I will simply update the docker image

**Comment by raul-arabaolaza** [comment]
Fix for the docker image in https://github.com/fbelzunc/active-directory-plugin-build-container-with-docker-fixtures/pull/1 waiting for @fbelzunc now. I have tested it locally and is able to run the tests without issues

**Comment by jglick** [comment]
> lacking permissions my changes on the Jenkinsfile would not be reflected

Hmm, https://issues.jenkins.io/browse/JENKINS-46795 again.

If you want to be a maintainer of the plugin, simply file a PR to RPU and I imagine we can get that sorted out.

> consolidate the `Dockerfile` in the plugin

Back to https://github.com/jenkinsci/active-directory-plugin/pull/142#issuecomment-1318733418? AFAICT you could just switch https://github.com/jenkinsci/active-directory-plugin/blob/be2f9e600e11981e1a5fb1f8fe56125db4fabf01/Jenkinsfile#L11-L16 to do something like

```groovy
 stage('maven') { 
    sh 'docker build -t fixture src/test/resources/fixture && docker run --add-host=samdom.example.com:127.0.0.1 -v /var/lib/docker --privileged --dns=127.0.0.1 --dns=8.8.8.8 -v $WORKSPACE:/project fixture clean install -P \'!noIT\'' 
 } 
```

**Comment by jglick** [comment]
Hmm, https://github.com/jenkins-infra/repository-permissions-updater/blob/58c9d82ed5f0af1d859447b856d6f85c21503526/permissions/plugin-active-directory.yml#L18 says you are already a maintainer, so if you happen to lack write permission to the repository, just https://github.com/jenkins-infra/helpdesk/issues/new?assignees=&labels=github-permissions%2Ctriage&template=3-github-permissions.yml

**Review by car-roll — APPROVED** [review]

**Comment by jglick** [comment]
@fbelzunc I guess you could archive https://github.com/fbelzunc/active-directory-plugin-build-container-with-docker-fixtures now.

**Pull request merged**